src/mercury_engine_data_structures/formats/bmsbk.py

```diff
@@ -67,3 +67,17 @@ class Bmsbk(BaseResource):
     @functools.lru_cache
     def construct_class(cls, target_game: Game) -> Construct:
         return BMSBK
+
+    def get_block_group(self, block_group: int) -> Container:
+        return self.raw.block_groups[block_group]
+
+    def set_block_type(self, block_group: int, block_type: str) -> Container:
+        weakness = self.get_block_group(block_group).types[0]
```

Comment:
What is this magical `0`? Is `types` always a 1-length vector? Maybe add some comments or asserts

Reply:
Yes types is always length 1. Might not be the best name for the field. This is how it looks for the entries.
```Container: 
            unk_bool = True
            types = ListContainer: 
                Container: 
                    block_type = u'power_beam' (total 10)
                    blocks = ListContainer: 
                        Container: 
                            pos = ListContainer: 
                                -1800.0
                                2000.0
                                0.0
                            unk2 = 0
                            unk3 = 0
                            respawn_time = 5.0
                            model_name = u'sg_casca111' (total 11)
                            vignette_name = u'' (total 0)
                        Container: 
                            pos = ListContainer: 
                                -1700.0
                                2000.0
                                0.0
                            unk2 = 0
                            unk3 = 0
                            respawn_time = 5.0
                            model_name = u'sg_casca112' (total 11)
                            vignette_name = u'' (total 0)

Reply:
Hm... I remember something in Dread about supporting multiple tile types/groups in a single "Block spawner" and it could have different types.

Easier to just not do that. The construct definition could just assert it's always size 1 and not use a list. Or as the simplest way, just add an assertion that `len(self.get_block_group(block_group).types) == 1` in case something funny happens.

Reply:
either is fine by mii